### ISSUE TITLE
Bus and Tramway line in colour

### ISSUE THREAD

**Issue by adrien-github** [osmandapp/Osmand#4787]
Hi,

I am using osmand 2.8 (from f-droit repo), with the offline maps.

In Paris (France), I can see that the metro line are displayed in colour, for instance 
the [Metro, line 12](http://www.openstreetmap.org/relation/123775#map=13/48.8654/2.3195&layers=D)

However, the bus and tramway lines are displayed only in red, for instance in Nantes the
[Tramway line 2 ](http://www.openstreetmap.org/relation/64032) or the [Bus Line 96](http://www.openstreetmap.org/relation/1710610)

All these 3 relations have the colour attribute.

It is possible to print the bus and tramway lines with their colour, instead of everything in red ?

Kind regards,

Adrien

**Comment by xmd5a2** [comment]
No, we don't plan to support different colours for all transport types except subway.

**Comment by prototype99** [comment]
would it be possible to at least have automatic generation to try and make sure the colours are different? (so not necessarily actively assigning colours to different bus lines etc. just making it so the renderer picks different colours for each line, based on what is currently on screen) in areas with no subway having the same colour for all buses would make all public transport unusable

**Comment by sonora** [comment]
@xmd5a2 Why would we not support the colour tag for bus relations? See #7466.

**Comment by xmd5a2** [comment]
@sonora This will lead to the need for a complete rework of all public transport scheme options because transport types differ one from another by hardcoded colours. Also we can't correctly display multiple routes on one way.

**Comment by vshcherb** [comment]
Well each color is supported and hardcoded separately in the style, it will slow down rendering and complicate rendering style a lot. There is not enough value for now to invest into it

**Comment by whyteks** [comment]
Doing some mapping of public transport routes gave me reason to examine this functionality in OsmAnd. 

There are a number of issues open in relation to colour of bus lines. 
https://github.com/osmandapp/OsmAnd/issues/3168
https://github.com/osmandapp/OsmAnd/issues/7466
https://github.com/osmandapp/OsmAnd/issues/12393

I choose this issue to add some thoughts, as I feel it may be useful to query some of the comments here, for clarity.

The problem of displaying multiple routes on a way is difficult, this may be a "UX" issue as much as a programming issue, and may require some creative thought. It might be interesting to see what a default colour for ways that have multiple routes, and colours where a single route exists - to see that that looks like.

All improvements of any program or application represents work, and somebody has to do it. If a complete rework of something is required, then this is because there is a problem, and the sooner it is dealt with, the better, IMHO.

It does seem hard to imagine that rendering bus lines in colour would represent a significant processing load. The comment is not so old, only 5 years, but all the same, today we have 3D rendering, hillshades and all kinds of CPU heavy stuff.

I have various devices, from 10 years old to 2 years old, and the older devices struggle to render. Still, It should be the choice of the user what to use their CPU cycles for, as much as it is also the choice of the user to decide what has "value". It is perfectly valid to want to prioritise visualisation of public transport routes. Here I might venture to summise that historically, (but not in all regions), a majority of effort into OSM mapping (and related such as OsmAnd) seems to goes into private transport routing, rather than public transport. So be it. Each to their own. I also use private transport routing in OsmAnd. This is no criticism, just an observation. We now so have experimental public transport routing in OsmAnd, which is an excellent addition, but with some issues, and maybe people interested in public transport need to step up and contribute. 

In some issues, people are confusing, or maybe better said, mixing realtime transport information with routing information. Thinking about it, if any realtime APIs were to be integrated, this is probably better left to the realm of specific plugins for networks, rather than core functionality, whereas colouring the bus routes, I believe, is apt for core functionality. 

I tried to see what I could do with editing the map rendering XML, but it seems, as somewhat expected, that we are limited here. I was able to change the halo colour for the route ref rendered on the way, and I thought this was interesting, as at least, maybe the bus routes colour info from OSM data could be integrated here. 

In the meantime, I have started digging into OSM code to see how much work this "rework" represents. I don't like to make "promises" about code that does not exist yet, we'll see how much time and energy I can find.. but it might be good to hear from you, if you're up for working on it too. Maybe there are more individuals with capability to code on OsmAnd who also have interest in public transport, or even interest in seeing improvements implemented.

Honest comments from experienced devs are also always useful. But please, not "No, won't implement, because I have no interest in this, or in my opinion it has no "value". 

This is FOSS, right? where one's own personal interests or opinions are not so relevant to what other people may or may not contribute.

Thanks!
